src/jit/block.h

```diff
@@ -1171,6 +1161,16 @@ struct BasicBlock : private LIR::Range
 
     void MakeLIR(GenTree* firstNode, GenTree* lastNode);
     bool IsLIR();
+
+    void SetDominatedByExceptionalEntryFlag()
+    {
+        bbFlags |= BBF_DOMINATED_BY_EXCEPTIONAL_ENTRY;
+    }
+
+    bool IsDominatedByExceptionalEntryFlag()
+    {
+        return (bbFlags & BBF_DOMINATED_BY_EXCEPTIONAL_ENTRY) != 0;
+    }
```

Comment:
So, the existing algorithm set this bit only on blocks that are dominated by a block from fgEnterBlks and this block is not normal entry.
So if any block was added after computing fgComputeEnterBlocksSet it was not reflected in fgEnterBlks and lclVars in this block were not marked with the hint.